packages/CatalogService/src/apiFactory/operations/utils/common.ts

```diff
@@ -124,44 +165,64 @@ export const ResourceSchema = z.union([
 
 export type ResourceType = z.infer<typeof ResourceSchema>;
 
-export const getAllRDFTriples = async (options: {
+export const getAllResources = async (options: {
   service: CatalogService;
   hasAccess?: boolean;
 }) =>
   RDFResponseSchema.parse(
     await options.service.runQuery(`
-      SELECT DISTINCT ?s ?p ?o
-      WHERE {
-        ${
-          options.hasAccess
-            ? // REQUIREMENTS 8.1 Search by user-owned data-resources
-              `?s dct:accessRights "${session.user.name}" .`
-            : ""
+    SELECT ?s ?p ?o WHERE {
+      {
+        SELECT ?s (<${RDF_TYPE_URI}> AS ?p) (<${DATASET_URI}> AS ?o) WHERE {
+          ?s <${RDF_TYPE_URI}> <${DATASET_URI}> .
+        }
+      } UNION {
+        SELECT ?s (<${RDF_TYPE_URI}> AS ?p) (<${SERVICE_URI}> AS ?o) WHERE {
+          ?s <${RDF_TYPE_URI}> <${SERVICE_URI}>  .
+        }
+      } UNION {
+        SELECT ?s (<${RDF_TYPE_URI}> AS ?p) (<${CATALOG_URI}> AS ?o) WHERE {
+          ?s <${RDF_TYPE_URI}> <${CATALOG_URI}> .
         }
-        ?s ?p ?o
-      }`)
+      }      
+    } 
+  `)
   );
 
-/**
- * Read UI data from the DCAT instances
- * @param options
- * @returns
- */
-export const uiDataResourceFromInstanceWithTriples =
-  (triples: RDFTripleType[]) =>
-  /**
-   *
-   * @param el
-   * @returns
-   */
-  async (
-    el: DCATDataset | DCATDataService | DCATCatalog
-  ): Promise<UIDataResourceType> => {
-    // Note: Renaming from spec-specific to spec-agnostic values
-    // Rationale: For now, keeping spec concepts out of UI
-    return UIDataResourceSchema.parse(
-      el.service.interpretation
-        ? await getValuesByCola(el, triples)
-        : await getValuesByBailey(el)
-    );
-  };
+export const getAllResourcesWithDetails = async (options: {
+  service: CatalogService;
+}) =>
+  ResourceDetailResponseSchema.parse(
+    await options.service.runQuery(`
+      SELECT ?s  ?p ?o ?relationship ?partner ?resourceTitle WHERE {
+        
+        {
+          SELECT ?s ?p ?o  WHERE {
+            {
+              SELECT ?s (<${RDF_TYPE_URI}> AS ?p) (<${DATASET_URI}> AS ?o) WHERE {
+                ?s <${RDF_TYPE_URI}> <${DATASET_URI}> .
+                
+              }
+            } UNION {
+              SELECT ?s (<${RDF_TYPE_URI}> AS ?p) (<${SERVICE_URI}> AS ?o) WHERE {
+                ?s <${RDF_TYPE_URI}> <${SERVICE_URI}>  .
+                
+              }
+            } UNION {
+              SELECT ?s (<${RDF_TYPE_URI}> AS ?p) (<${CATALOG_URI}> AS ?o) WHERE {
+                ?s <${RDF_TYPE_URI}> <${CATALOG_URI}> .
+               
+              }
+            }      
+          }
+        }
+        OPTIONAL {
+          ?s ?relationship ?partner .
+          FILTER(?relationship in (<${DATASET_PROP_URI}>, <${SERVICE_PROP_URI}>, <${CATALOG_PROP_URI}>))
+        }
+        OPTIONAL {
+          ?s dct:title ?resourceTitle .
+        }
+      } 
+    `)
```

Comment:
Idea: Some important parts of this might benefit from a look-in from the DATA team